app/models/hardware.rb

```diff
@@ -19,7 +19,7 @@ class Hardware < ApplicationRecord
   has_many    :guest_devices, :dependent => :destroy
   has_many    :storage_adapters, -> { where "device_type = 'storage'" }, :class_name => "GuestDevice", :foreign_key => :hardware_id
   has_many    :nics, -> { where "device_type = 'ethernet'" }, :class_name => "GuestDevice", :foreign_key => :hardware_id
-  has_many    :ports, -> { where "device_type != 'storage'" }, :class_name => "GuestDevice", :foreign_key => :hardware_id
+  has_many    :ports, -> { where "device_type = 'ethernet port'" }, :class_name => "GuestDevice", :foreign_key => :hardware_id
```

Comment:
I'm worried about this, but without any evidence...

The new implementation is much more "correct".  I'm just worried that there were some old assumptions built into that previous relationship mapping based on `!= 'storage'` that will come back to bite us.

Honestly, the only thing I can think of doing is letting it through and crossing our fingers.